enhancements/console/dynamic-plugins.md

```diff
@@ -147,6 +147,44 @@ spec:
     basePath: '/'
 ```
 
+In case the plugin needs to communicate with an in-cluster services, it can
+declare the communication in it's `ConsolePlugin` manifest, defining
+`spec.proxyService` array field entry. In there service `name`,
+`namespace`, `port` and `basePath` needs to specified. The console backend
+will proxy the plugin request to the dependent service using the service
+CA bundle.
+
+After that plugin can make request to the console endpoint, which will
+proxy the request to the dependent service. Plugin's request shall use
+following console's endpoint:
+`/api/plugin-proxy/<dependent-service-name>/<request-path>`
+
+In case the service is using a custom service-ca, `caCertificate` field
+for the service can be specified. In case the the service proxy request
+shall contain user's authentication token, the `authorize` field needs
+to be set to `true`, otherwise the token won't be part of the proxy request.
+
+```yaml
+apiVersion: console.openshift.io/v1alpha1
+kind: ConsolePlugin
+metadata:
+  name: my-plugin
+spec:
+  displayName: 'My Plugin'
+  service:
+    name: my-console-plugin
+    namespace: my-operator-namespace
+    port: 8443
+    basePath: '/'
+  proxyService:
+  - name: plugin-dependent-service
+    namespace: plugin-dependent-service-namespace
+    port: 8443
+    basePath: '/'
+    caCertificate: '-----BEGIN CERTIFICATE-----\nMIID....'
```

Comment:
Typically this would be a config map reference instead of specified inline.

Reply:
Right, I just saw that some structs are using inlined CA certs so wanted to give it a try.
Also wanted to avoid granting console || console-operator any additional rights, otherwise the CM would need to be created in the `openshift-config-managed` || `openshift-config` (depends on the implementation). Or we could create a separate NS for this but not sure if thats not an overkill... 
Thoughts ?

Reply:
route API is inlining the cert https://github.com/openshift/api/blob/master/route/v1/types.go#L234